src/Application/GameLuaBindings.cpp

```diff
@@ -90,18 +125,64 @@ void GameLuaBindings::init(lua_State *L) {
             SetCurrentMenuID(MenuType(screenIndex));
         }
     );
+}
+
+void _registerAudioBindings(sol::state_view &luaState, sol::table &table) {
+    table["audio"] = luaState.create_table_with(
+        "playSound", [](SoundId soundId, SoundPlaybackMode mode) {
+            pAudioPlayer->playSound(soundId, mode);
+        },
+        "playMusic", [](MusicId musicId) {
+            pAudioPlayer->MusicPlayTrack(musicId);
+        }
+    );
+}
+
+void _registerItemBindings(sol::state_view &luaState, sol::table &table) {
+    typedef std::function<bool(ItemId)> FilteItemFunction;
+
+    auto createItemTable = [&luaState](const ItemDesc &itemDesc) {
+        return luaState.create_table_with(
+            "name", itemDesc.name,
+            "level", itemDesc.uItemID_Rep_St
+        );
+    };
+
+    table["items"] = luaState.create_table_with(
+        "get_item_info", [&luaState, createItemTable](ItemId itemId) {
+            if(itemId >= ITEM_FIRST_VALID && itemId <= ITEM_LAST_VALID) {
+                const ItemDesc &itemDesc = pItemTable->pItems[itemId];
+                return sol::object(luaState, createItemTable(itemDesc));
+            }
+            return sol::make_object(luaState, sol::lua_nil);
+        },
+        // The get_random_item function accept an optional filter function to exclude some items from the randomization
+        "get_random_item", [](const FilteItemFunction &filter) {
+            if(filter) {
+                std::vector<ItemId> itemsToRandomizeOn;
+                Segment<ItemId> &&spawnableItems = allSpawnableItems();
+                for (ItemId itemId : spawnableItems) {
```

Comment:
`for (ItemId itemId : allSpawnableItems())` should work.

If you want to make it terser then:
```
for (ItemId itemId : allSpawnableItems() | std::views::filter(filter)) {
    itemsToRandomizeOn.push_back(itemId);
}
```

Reply:
Uuh nice, sure! I should start using range and views or I'll never learn about them :D